display_list/display_list_image_filter.h

```diff
@@ -137,6 +153,26 @@ class DlImageFilter
                                            const SkMatrix& ctm,
                                            SkIRect& input_bounds) const = 0;
 
+  MatrixCapability get_matrix_capability() const {
+    MatrixCapability result = this->matrix_capability();
+    // CropRects need to apply in the source coordinate system, but are not
+    // aware of complex CTMs when performing clipping. For a simple fix, any
+    // filter with a crop rect set cannot support more than scale+translate CTMs
+    // until that's updated. if (this->cropRectIsSet()) {
+    //     result = std::min(result, MatrixCapability::kScaleTranslate);
+    // }
+    for (auto* filter : filters()) {
```

Comment:
This feels a little like overkill for our one type that has sub-filters. I would just make this virtual and have Compose return "min(outer->..., inner->...).

Reply:
Done.
Another question is the Skia has a logic is check the `cropRectIsSet`, sorry I don't get what it means.

Should we need to add this check?

Reply:
If it's the code I think you are referring to, I think they have a way to inject a crop rectangle on the inputs that we don't necessarily have (?yet?)?